File: src/types/dispensing.ts
import type { DoseSchedule } from '#/types/visit'

export interface ParsedDoseInfo {
  tabletsPerDose: number
  mgPerDose: number
  mgPerWeek: number
  mgPerDayAverage: number
  schedule: DoseSchedule
  matchedDays: string[]
}

export interface DispensingRecord {
  hn: string
  vn?: string
  vstdate: string
  icode: string
  drugName: string
  strength: string
  qty: number
  unitprice: number
  drugusageCode?: string
  spUseCode?: string
  usageText?: string
  parsedDose?: ParsedDoseInfo
  usageParseNote?: string
}

File: src-tauri/src/models/dispensing.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct DispensingRecord {
  pub hn: String,
  pub vstdate: String,
  pub icode: String,
  pub drug_name: String,
  pub strength: String,
  pub qty: f64,
  pub unitprice: f64,
}
